pkg/controller/jobs/job/job_controller.go

```diff
@@ -380,3 +407,80 @@ func SetupIndexes(ctx context.Context, fieldIndexer client.FieldIndexer) error {
 func GetWorkloadNameForJob(jobName string, jobUID types.UID) string {
 	return jobframework.GetWorkloadNameForOwnerWithGVK(jobName, jobUID, gvk)
 }
+
+func isRaySubmitterJobWithAutoScaling(ctx context.Context, jobObj client.Object, k8sClient client.Client) (bool, *rayv1.RayJob) {
```

Comment:
nit. please move the helpers to a dedicated file in this package: `ray_utils.go`, and tests to `ray_utils_test.go`